buildSrc/src/main/kotlin/terasology-module.gradle.kts

```diff
@@ -100,21 +83,49 @@ configure<SourceSetContainer> {
     }
 }
 val convention = project.getConvention().getPlugin(JavaPluginConvention::class)
-val mainSourceSet = convention.getSourceSets().getByName("main");
+val mainSourceSet = convention.getSourceSets().getByName("main")
 
 // TODO: Remove when we don't need to rely on snapshots. Needed here for solo builds in Jenkins
 configurations.all {
     resolutionStrategy.cacheChangingModulesFor(0, "seconds")
 }
 
+
+val deps = moduleConfig.dependencies.filterNotNull()
+val moduleDepends = deps.filterNot { it.id.toString() == "engine" }
+val engineVersion = deps.find { it.id.toString() == "engine" }?.versionRange()?.toString() ?: "+"
+
 // Set dependencies. Note that the dependency information from module.txt is used for other Terasology modules
 dependencies {
-    // Check to see if this module is not the root Gradle project - if so we are in a multi-project workspace
-    implementation(group = "org.terasology.engine", name = "engine", version = "+") { isChanging = true }
-    implementation(group = "org.terasology.engine", name = "engine-tests", version = "+") { isChanging = true }
+    implementation(group = "org.terasology.engine", name = "engine", version = engineVersion) { isChanging = true }
+    implementation(group = "org.terasology.engine", name = "engine-tests", version = engineVersion) { isChanging = true }
+
+    for (gestaltDep in moduleDepends) {
+        if (!gestaltDep.minVersion.isSnapshot) {
+            // gestalt considers snapshots to satisfy a minimum requirement:
+            // https://github.com/MovingBlocks/gestalt/blob/fe1893821127/gestalt-module/src/main/java/org/terasology/naming/VersionRange.java#L58-L59
+            gestaltDep.minVersion = gestaltDep.minVersion.snapshot
+            // (maybe there's some way to do that with a custom gradle resolver?
+            // but making a resolver that only works that way on gestalt modules specifically
+            // sounds complicated.)
+        }
 
-    for (dependency in moduleDepends) {
-        implementation(group = "org.terasology.modules", name = dependency, version = "+") { isChanging = true }
+        val gradleDep = create(
+            group = "org.terasology.modules",
+            name = gestaltDep.id.toString(),
+            version = gestaltDep.versionRange().toString()
+        )
+
+        if (gestaltDep.isOptional) {
+            // `optional` module dependencies are ones it does not require for runtime
+            // (but will use opportunistically if available)
+            compileOnly(gradleDep) { isChanging = true }
+            // though modules also sometimes use "optional" to describe their test dependencies;
+            // they're not required for runtime, but they *are* required for tests.
+            testImplementation(gradleDep) { isChanging = true }
```

Comment:
We should probably introduce pure test dependencies in module land... a task for another day ^^